libs/SalesforceSDKCommon/SalesforceSDKCommon/Classes/Logger/SFLogger.m

```diff
@@ -65,84 +103,106 @@ -(SFLogLevel)logLevel {
 -(void)setLogLevel:(SFLogLevel) logLevel {
     [self.logger setLogLevel:logLevel];
 }
+
+- (void)submitLogEntryWithClass:(nonnull Class)cls level:(SFLogLevel)level message:(nonnull NSString *)message {
+    [self.logger log:cls level:level message:message];
+    [self.logReceiver receiveWithLevel:level cls:cls component:self.logger.componentName message:message];
+}
+
+/**
+ * Submits a log entry with the provided properties to the logger and log receiver.
+ */
+- (void)submitLogEntryWithClass:(nonnull Class)cls level:(SFLogLevel)level format:(nonnull NSString *)format args:(va_list)args {
+    [self.logger log:cls level:level format:format args:args];
+    NSString *formattedMessage = [[NSString alloc] initWithFormat:format arguments:args];
+    [self.logReceiver receiveWithLevel:level cls:cls component:self.logger.componentName message:formattedMessage];
+}
+
 - (void)log:(nonnull Class)cls message:(nonnull NSString *)message {
-    [self.logger log:cls level:SFLogLevelDefault message:message];
+    [self submitLogEntryWithClass:cls level:SFLogLevelDefault message:message];
 }
 
 - (void)log:(nonnull Class)cls format:(nonnull NSString *)format, ... {
     va_list args;
     va_start(args, format);
-    [self.logger log:cls level:SFLogLevelDefault format:format args:args];
+    [self submitLogEntryWithClass:cls level:SFLogLevelDefault format:format args:args];
     va_end(args);
 }
 
 - (void)log:(Class)cls level:(SFLogLevel)level message:(NSString *)message {
-    [self.logger log:cls level:level message:message];
-}
+    [self submitLogEntryWithClass:cls level:level message:message];}
 
 - (void)log:(Class)cls level:(SFLogLevel)level format:(NSString *)format, ... {
     va_list args;
     va_start(args, format);
-    [self.logger log:cls level:level format:format args:args];
+    [self submitLogEntryWithClass:cls level:level format:format args:args];
     va_end(args);
 }
 
 - (void)log:(Class)cls level:(SFLogLevel)level format:(NSString *)format args:(va_list)args {
+    va_list args_copy;
+    va_copy(args_copy, args); // Note: This log convenience cannot use the common submit log entry methods since it requires this local memory copy due to the caller already having run `va_start` on `args`.
```

Comment:
@bbirman - I added a new comment here so future maintainers will know why this one method still has its own implementation.